src/main/java/core/basesyntax/dao/WriteToFile.java

```diff
@@ -0,0 +1,56 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.products.Fruit;
+import core.basesyntax.storage.ListStorage;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class WriteToFile {
+    public boolean writeToFile(String filePath) {
+        Path path = Paths.get(filePath);
+        if (Files.exists(path)) {
+            try {
+                Files.delete(path);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+            createFile(filePath);
+        } else {
+            createFile(filePath);
+        }
+
+        Map<String, Integer> mapToFile = ListStorage.listStorage.stream()
+                .collect(Collectors.groupingBy(Fruit::getName,
+                        Collectors.summingInt(Fruit::getAmount)));
```

Comment:
This is violation of SRP. This class, even this method, not only writes data to file - it always perform some business logic doing action to create data for writing to file.

Reply:
Ok, agree made one more class, moved it there, and prepared a test for this class.